### PR TITLE
New app: io.github.twanvl.MagicSetEditor2

### PR DESCRIPTION
**Pull request by subpop**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.twanvl.MagicSetEditor2.desktop [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Type=Application
+Version=1.0
+Name=MagicSetEditor
+Comment=A program used to make custom cards
+Exec=/app/bin/magicseteditor
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.desktop:6]:
```suggestion
Exec=magicseteditor
```

File: io.github.twanvl.MagicSetEditor2-64.png
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2-64.png:1]: you don't need a 64px icon if it doesn't come from the upstream sources.

File: io.github.twanvl.MagicSetEditor2.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>GPL-2.0</project_license>
+	<url type="homepage">https://github.com/twanvl/MagicSetEditor2</url>
+	<releases>
+	  <release version="2.1.2" date="2022-09-22" />
+	</releases>
+	<content_rating type="oars-1.1">
+    <content_attribute id="violence-fantasy">mild</content_attribute>
+  </content_rating>
+</component>
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.metainfo.xml:19]: missing mandatory screenshots.

File: io.github.twanvl.MagicSetEditor2.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+sdk: org.freedesktop.Sdk
+command: magicseteditor
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --filesystem=home
+  - --filesystem=~/.magicseteditor
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.yaml:13]: this should be 
```suggestion
  - --persist=.magicseteditor
```

File: io.github.twanvl.MagicSetEditor2.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+app-id: io.github.twanvl.MagicSetEditor2
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: magicseteditor
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.yaml:10]: if it really supports wayland then it should be fallback-x11

File: io.github.twanvl.MagicSetEditor2.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      - type: archive
+        url: https://downloads.sourceforge.net/boost/boost/1.75.0/boost_1_75_0.tar.bz2
+        sha256: 953db31e016db7bb207f11432bef7df100516eeb746843fa0486a222e3fd49cb
+    build-commands:
+      - ./bootstrap.sh --prefix=${FLATPAK_DEST} --with-libraries=regex
+      - ./b2 headers
+      - ./b2 -j$FLATPAK_BUILDER_N_JOBS install variant=release cxxstd=17 --layout=system
+  - name: magicseteditor
+    buildsystem: cmake
+    sources:
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.yaml:45]: (everywhere) sources is usually last.

File: io.github.twanvl.MagicSetEditor2.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        path: io.github.twanvl.MagicSetEditor2.desktop
+      - type: file
+        path: io.github.twanvl.MagicSetEditor2.metainfo.xml
+      - type: file
+        path: io.github.twanvl.MagicSetEditor2-64.png
+      - type: file
+        path: io.github.twanvl.MagicSetEditor2-128.png
+    no-make-install: true
+    post-install:
+      - install -D --mode=755 ./magicseteditor /app/bin/
```
Comment (hfiguiere) [inline io.github.twanvl.MagicSetEditor2.yaml:59]:
```suggestion
      - install -Dm755 ./magicseteditor -t /app/bin/
```

**Comment by hfiguiere** [comment]
bot, build io.github.twanvl.MagicSetEditor2

**Review by hfiguiere — CHANGES_REQUESTED** [review]
the desktop, appstream and icon files should be submitted upstream.

**Comment by barthalion** [comment]
There is no upstream activity since 2021 so I assume it's fine not to upstream anything.

**Comment by subpop** [comment]
Yea, upstream is not dead, but it's extremely inactive. This is a program that has been limping along for the better part of a decade or more already. This upstream is a "keep-alive" fork as it is. I'll submit some of these changes upstream, but I don't expect them to be accepted.

**Comment by subpop** [comment]
bot, build io.github.twanvl.MagicSetEditor2

**Comment by subpop** [comment]
bot, build io.github.twanvl.MagicSetEditor2

**Comment by subpop** [comment]
bot, build io.github.twanvl.MagicSetEditor2

**Pull request closed without merging**